### PR TITLE
Centos support new

### PR DESCRIPTION
**Pull request by Ykisialiou**
New version of centos support implementation.

The things are covered is almost the same.

- [x]  `hosts.example` changed, centos and ubuntu groups added
- [x] preconf role chrony tasks refactored #130 
- [x] preconf role packages tasks refactored
- [x] nodejs role refactoring cherry-picked
- [x] nginx role refactoring cherry-picked
- [x] debug-tools role refactored
- [x] minor python3 related bugs fixed
- [x] nginx openssl dependencies refactored #90

### CODE REVIEW

**Comment by phahulin** [comment]
I have a question about organizing variables:

- we use `group_vars/ubuntu` and `group_vars/centos` to separate operating systems, but on the other hand, there are parts of code where we rely on `ansible_os_family == "Debian" ` and `ansible_os_family == "RedHat" `. Can we use `ansible_os_family` check everywhere? Store os-related variables like `ansible_python_interpreter` in different files and load them depending on `ansible_os_family`

**Comment by phahulin** [comment]
There are 4 more variables that are OS-dependent:

* ubuntu:
```
PARITY_BIN_LOC: "https://s3.us-east-2.amazonaws.com/poa-builds-parity-published/1.10.6/parity"
PARITY_BIN_SHA256: "539f4788fbd605a9cd87b5bf747b27ae05b8a4080b26aa3da645b0446fa9f9cc"
ORCHESTRATOR_BIN_LOC: ""
ORCHESTRATOR_BIN_SHA256: ""
```

* centos
```
PARITY_BIN_LOC: "https://releases.parity.io/v1.10.6/x86_64-unknown-centos-gnu/parity"
PARITY_BIN_SHA256: "9dd5495cadc39c90b1e40a7b3a449821bab84e55d489a688cf5b1c68c4852254"
ORCHESTRATOR_BIN_LOC: ""
ORCHESTRATOR_BIN_SHA256: ""
```

`ORCHESTRATOR_BIN_*` are the same (empty) for now, but when orchestrator (different project) is ready they will be different

File: roles/nodejs/tasks/main.yml [hunk 1]
```diff
@@ -1,17 +1,6 @@
...
 
-- name: add nodejs repository key
-  apt_key:
-    keyserver: keyserver.ubuntu.com
-    id: 68576280
-    state: present
-
-- name: Add nodejs repository
-  apt_repository:
-    repo: 'deb {{ NODE_SOURCE_DEB }} {{ ansible_distribution_release }} main'
```
[the remark this answers is not in the document]
Comment (Ykisialiou) [reply roles/nodejs/tasks/main.yml]: - [x] group_vars/ubuntu and group_vars/centos are left as they are
 we discussed this in Slack
- [x] `.example` suffix removed from group_vars/ubuntu and group_vars/centos as they have only defaults inside that will be rare or even never modified 
- [x] `NODE_SOURCE_DEB` left only in role variables

File: roles/nodejs/tasks/centos.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+---
+- name: Import Nodesource RPM key
+  rpm_key:
+    key: "{{ epel_key_url }}"
+    state: present
+
+- name: Add Nodesource repositories for Node.js
+  yum:
+    name: "https://rpm.nodesource.com/pub_8.x/el/{{ ansible_distribution_major_version }}/{{ ansible_architecture }}/nodesource-release-el{{ ansible_distribution_major_version }}-1.noarch.rpm"
```
[the remark this answers is not in the document]
Comment (Ykisialiou) [reply roles/nodejs/tasks/centos.yml]: Done. ( de777036b1cd6952c58e903b1b99c56ae8e13498 )
I hope it's what was requeted. yum demands direct link to rpm package in this case, so it's now formed in `roles/nodejs/vars/main.yml`

[the remark this answers is not in the document]
Comment (phahulin) [reply roles/nodejs/tasks/centos.yml]: Yes, that's what I meant, thanks

**Comment by phahulin** [comment]
1. can ubuntu/centos distinction be also made for playbook in `aws` subfolder, i.e. for variables
`image` and `user`? So that user could create a CentOS box. There is no hosts file in that playbook, so I'm not sure if that's possible

2. please remove `ansible_user`, `ansible_python_interpreter` from `group_vars/all.yml.network` since both of them moved to `group_vars/{ubuntu,centos}` now

3. I ran into the following error:
```

TASK [preconf : add swap file to system] ******************************************************************************************************************************************************************************************************
fatal: [54.221.7.136]: FAILED! => {"changed": true, "cmd": "/sbin/swapon /swapfile", "delta": "0:00:00.036229", "end": "2018-06-29 12:17:30.211268", "msg": "non-zero return code", "rc": 255, "start": "2018-06-29 12:17:30.175039", "stderr": "swapon: /swapfile: swapon failed: Invalid argument", "stderr_lines": ["swapon: /swapfile: swapon failed: Invalid argument"], "stdout": "", "stdout_lines": []}
```

looks like there are some issues with using `fallocate` for swap in CentOS https://unix.stackexchange.com/questions/294600/i-cant-enable-swap-space-on-centos-7

I tried solution suggested in the link above and replaced https://github.com/Ykisialiou/deployment-playbooks/blob/e883437ad644fdfcbe9bfb26778b76827c4da768/roles/preconf/tasks/swap.yml#L7-L11 with
```
#for centos
- name: create file for swap
  shell: dd if=/dev/zero of=/swapfile count=1024 bs=1MiB
  when: not swap_file_check.stat.exists
  tags:
    - skip_ansible_lint
```
and it seem to have worked, but it needs to be properly done so that it applies only for centos.

4. then I got the following error at the end of play
```
RUNNING HANDLER [preconf : restart chrony] ****************************************************************************************************************************************************************************************************
fatal: [54.221.7.136]: FAILED! => {"changed": false, "msg": "Could not find the requested service chrony: host"}
```
It seems like in ubuntu this service is called `chrony` while in centos it's called `chronyd`, so both should be declared in `handlers/`

**Comment by phahulin** [comment]
Should `aws/group_vars/all.yml.example` also contain `access_key` and `secret_key` for aws user credentials as `group_vars/all.yml` does?
Otherwise LGTM, so I will approve it

**Comment by phahulin** [comment]
Also, please add `aws/group_vars/all.yml` to `.gitignore`

**Pull request merged**